### DISCUSSION TITLE
Skeleton3D Multi-Bone Operations

### DISCUSSION THREAD

**Discussion by SoapSpangledGames** [godotengine/godot-proposals#4966] [Editor]
Godot 4's animation system has come a long way, and is largely useful. However, it has one major show stopper: it doesn't allow for operations to be performed on more than one bone at a time.

In order to animate a crouch (which is used for multiple purposes), the bones for the entire upper body need to be moved down and up for different keyframes, but Godot's bone animation system requires that each bone be moved independently of every other bone. This is very cumbersome, makes the entire animation system difficult to use despite how well the rest of it works (and the rest of it is a godsend!), and makes it nearly impossible to maintain bone spatial relationships after bone translations.

It would be very useful to be able to select and manipulate multiple bones at a time, and is so far the only thing I've encountered that keeps me from dropping Blender for character animations. Using Blender for character animations causes premature aging and death, so it would be so awesome to be able to do it all from within Godot.

**Comment by TokageItLab** [comment]
I know there is still room for improvement in SkeletonGizmo. I think it would be difficult to do that in 4.0, but adding things like multiple selections and independent Transforms makes sense if we want to create animations on Godot. However, to create better animations, things like IK are necessary, so we will need to think about how to interact with them as well.
